qutip/solver/solver_base.py

```diff
@@ -39,12 +39,13 @@ class Solver:
         "normalize_output": "ket",
         "method": "adams",
     }
+    _reset_options = {"method"}
     resultclass = Result
 
     def __init__(self, rhs, *, options=None):
         if isinstance(rhs, (QobjEvo, Qobj)):
             self.rhs = QobjEvo(rhs)
-        else:
+        elif rhs is not None:
```

Comment:
All subclasses of `Solver` should have one path for building the RHS. Having two paths makes the code much harder to reason about.

I think `_build_rhs()` is a good idea, although I think we should implement it on all sub-classes of `Solver` and have `Solver` do `self.rhs = self._build_rhs()` at the end of `__init__` and when options are changed (if needed). That way `.rhs` is always correct for the given options.